### DISCUSSION TITLE
How to make unit tests using Refit

### DISCUSSION THREAD

**Discussion by TortSweet** [reactiveui/refit#1386] [Q&A] [answered]
Hello everybody,
how can i test my service that used refit interface. I trying to mock my Refit interface, but than i run the test i have a problem: 

> System.AggregateException: 
> One or more errors occurred. (The connection is not established because the destination computer rejected the connection request. (localhost:7182)) ---> System.Net.Http.Http.HttpRequestException: The connection is not established because the endpoint rejected the connection request. (localhost:7182) ---> System.Net.Sockets.SocketException: The connection is not established because the end computer rejected the connection request.

How can I solve this problem?
Please share your opinion

**Comment by anaisbetts** [comment, marked as answer]
Generally you just mock the interface you are auto-generating against, you don't need to mock Refit itself

**Reply by bednart** [reply]
Hi,
what if result type from interface method is Task<ApiResponse<T>. If I try to setup mock with ReturnsAsync it fails with exception

Can not create proxy for method System.Threading.Tasks.Task`1[Refit.IApiResponse`1[System.String]] SendEmail(SendEmailRequest, System.Threading.CancellationToken) because it or its declaring type is not accessible. 

Is it possible to UnitTest ApiResponse with xunit?

EDIT:
Problem solved. It works but interface with refit methods has to be public. With internal it fails.
